kong/api/routes/certificates.lua

```diff
@@ -3,74 +3,77 @@ local utils = require "kong.tools.utils"
 local cjson = require "cjson"
 
 
-return {
-  ["/certificates/"] = {
-    POST = function(self, dao_factory, helpers)
-      local snis
-      if type(self.params.snis) == "string" then
-        snis = utils.split(self.params.snis, ",")
-        self.params.snis = nil
+local function create_certificate(self, dao_factory, helpers)
+  local snis
+  if type(self.params.snis) == "string" then
+    snis = utils.split(self.params.snis, ",")
+    self.params.snis = nil
+  end
+
+  if snis then
+    -- dont add the certificate or any snis if we have an SNI conflict
+    -- its fairly inefficient that we have to loop twice over the datastore
+    -- but no support for OR queries means we gotsta!
+    local seen = {}
+
+    for _, sni in ipairs(snis) do
+      if seen[sni] then
+        return helpers.responses.send_HTTP_CONFLICT(
+          "duplicate requested sni name " .. sni
+        )
       end
 
-      if snis then
-        -- dont add the certificate or any snis if we have an SNI conflict
-        -- its fairly inefficient that we have to loop twice over the datastore
-        -- but no support for OR queries means we gotsta!
-        local seen = {}
+      local cnt, err = dao_factory.ssl_servers_names:count({
+        name = sni,
+      })
+      if err then
+        return helpers.yield_error(err)
+      end
 
-        for _, sni in ipairs(snis) do
-          if seen[sni] then
-            return helpers.responses.send_HTTP_CONFLICT(
-              "duplicate requested sni name " .. sni
-            )
-          end
+      if cnt > 0 then
+        return helpers.responses.send_HTTP_CONFLICT(
+          "entry already exists with name " .. sni
+        )
+      end
 
-          local cnt, err = dao_factory.ssl_servers_names:count({
-            name = sni,
-          })
-          if err then
-            return helpers.yield_error(err)
-          end
+      seen[sni] = true
+    end
+  end
 
-          if cnt > 0 then
-            return helpers.responses.send_HTTP_CONFLICT(
-              "entry already exists with name " .. sni
-            )
-          end
+  local ssl_cert, err = dao_factory.ssl_certificates:insert(self.params)
+  if err then
+    return helpers.yield_error(err)
+  end
 
-          seen[sni] = true
-        end
-      end
+  -- insert SNIs if given
 
-      local ssl_cert, err = dao_factory.ssl_certificates:insert(self.params)
+  if snis then
+    ssl_cert.snis = {}
```

Comment:
I noticed that you added the `cjson.empty_array_mt` metatable to the `PUT` handler, but not this one. A better approach might be to leave both tables without the metatable, and include the metatable fix in another commit/PR to keep the scope of this one narrower if you would.

Reply:
Actually, we had fixed most of the `cjson.empty_array_mt` fixes in this file in https://github.com/Kong/kong/commit/39996523b3e7f387a4d9562e2d6274c4f0082b5f.

I'll fix this remaining one with this PR itself.